packages/core/database/src/metadata/metadata.ts

```diff
@@ -93,7 +95,14 @@ export class Metadata extends Map<string, Meta> {
   }
 }
 
-const createAttribute = (attributeName: string, attribute: Attribute) => {
-  const columnName = identifiers.getColumnName(attributeName);
+const createAttribute = (attributeName: string, attribute: Attribute, options: MetadataOptions) => {
+  // if the attribute has already set its own column name, use that
+  // this will prevent us from shortening a name twice
+  if ('columnName' in attribute && attribute.columnName) {
```

Comment:
Nice catch

Reply:
So difficult to debug and find a solution for, and yet so simple once I found it 😆